src/Surfnet/StepupSelfService/SelfServiceBundle/Service/RemoteVetting/Encryption/IdentityEncrypter.php

```diff
@@ -40,26 +41,18 @@ public function __construct(RemoteVettingConfiguration $configuration, IdentityW
     }
 
     /**
-     * @param IdentityDto $identity
-     * @param string $source
+     * @param AttributeListDto $identity
+     * @throws \Exception
      */
-    public function encrypt(IdentityDto $identity, $source)
+    public function encrypt(AttributeListDto $identity)
     {
-        $data = $this->constructData($identity, $source);
+        $data = json_encode($identity->jsonSerialize());
         $publicKey = $this->configuration->getPublicKey();
 
-        $encryptedData = '';
-        openssl_public_encrypt($data, $encryptedData, $publicKey);
-        $this->writer->write($encryptedData);
-    }
+        $encrypter = new XMLSecurityKey(XMLSecurityKey::AES256_CBC);
+        $encrypter->loadKey($publicKey, false, true);
+        $encryptedData = $encrypter->encryptData($data);
 
-    private function constructData(IdentityDto $identityDto, $source)
-    {
-        return json_encode(
-            array_merge(
-                $identityDto->jsonSerialize(),
-                ['version' => $this->configuration->getVersion(), 'source' => $source]
-            )
-        );
+        $this->writer->write($encryptedData);
```

Comment:
I removed it from the Encrypter so this could be later added to the AttributeListDto. That way Encrypter::encrypt could only be supplied with an object implementing the Serializable interface and that way a lot of coupling is removed.

Reply:
Okay